### PR TITLE
[AGM-1958] Update filter policy

### PR DESCRIPTION
**Pull request by vbagirov**

### CODE REVIEW

**Comment by slavaleleka** [comment]
linter check failed

File: docs/general/ad-filtering/filter-policy.md [hunk 1]
```diff
@@ -3,123 +3,291 @@ title: AdGuard filter policy
...
+**Ad blocker** is a software or hardware tool designed to filter, hide, or block advertisements and other elements on web pages. Ad blockers are created to improve user experience, reduce page load times, decrease internet traffic consumption, block ads, enhance privacy while browsing websites and using applications, and block annoying elements.
 
-## Ad filters
 
-This part describes the AdGuard Base filter, the Mobile Ads filter, and the following language-specific filters: Chinese, Dutch, French, German, Japanese, Russian, Spanish/Portuguese, and Turkish.
+A **filter** (or a **filter list**) is a set of filtering rules designed to filter content and block various types of advertisements and other types of content using ad blockers. These filters usually operate at the level of the web browser, programs, or DNS server.
```
Comment (Alex-302) [inline docs/general/ad-filtering/filter-policy.md:29]: лишняя пустая строка
```suggestion
```

File: docs/general/ad-filtering/filter-policy.md [hunk 2]
```diff
@@ -3,123 +3,291 @@ title: AdGuard filter policy
...
+- Anti-adblock scripts that prevent site usage (see *Limitations and Exceptions*)
+- Advertising injected by malware, if detailed information about its loading method or steps for reproduction is provided
+- Unwanted mining — scripts that mine cryptocurrency without user consent
 
-**Limitations and Exceptions**
+### Limitations and exceptions
+
+- The site’s own advertising should not be deliberately blocked. However, it should not be unblocked if the blocking is caused by general filtering rules
+- Content access measures like paywalls are not blocked
+- Anti-adblock walls will be blocked in the following cases:
```
Comment (Alex-302) [inline docs/general/ad-filtering/filter-policy.md:76]: Лишний пробел
```suggestion
- Anti-adblock walls will be blocked in the following cases:
```

File: docs/general/ad-filtering/filter-policy.md [hunk 3]
```diff
@@ -3,123 +3,291 @@ title: AdGuard filter policy
...
+- AdGuard Widgets filter
+- AdGuard Other Annoyances filter
+- AdGuard Annoyances filter — a combined filter that includes all 5 specialized annoyance filters
 
-If a rule is subject to the list of limitations described below, then it won’t be added to this filter.
+### What will be blocked with these filters?
 
-- Rules that cause problems with website functionality will be removed.
-- Website-specific rules will be added only if a website has sufficient traffic. Traffic is determined by open statistics (if available) or by other means, such as followers on social media. A website's traffic is considered sufficient when it has 100 thousand visits per month. We will consider adding a rule for a website that's not that popular, but the final decision is up to the filter list maintainer.
+#### AdGuard Cookie Notices filter
```
Comment (Alex-302) [inline docs/general/ad-filtering/filter-policy.md:156]: пробел в конце
```suggestion
#### AdGuard Cookie Notices filter
```

File: docs/general/ad-filtering/filter-policy.md [hunk 4]
```diff
@@ -3,123 +3,289 @@ title: AdGuard filter policy
...
-- Anti-adblock scripts will be blocked only if they limit or affect the functionality of a website.
-- Anti-adblock scripts will not be blocked in cases when it is prohibited by law.
-- Rules that often cause problems with the work of some websites will be deleted.
+### What these filters do
 
-## Tracking Protection filter
+- **Base filter** is designed to block ads on English-language websites and those for which there is no separate filter. It also contains general filtering rules that apply to all sites regardless of language.
+- **Mobile Ads filter** blocks advertisements on mobile versions of websites and within mobile applications. There is no segmentation based on language.
+- **Regional filters** follow the same policy as the **Base filter**, but limited to websites in certain languages.
```
Comment (piquark6046) [inline docs/general/ad-filtering/filter-policy.md:49]: If there is a website that uses a language [^1] and an advert issue exists with only its mobile version, can the issue be addressed in the language-specific filters list?

[^1]: AdGuard has a filters list for the language.

File: docs/general/ad-filtering/filter-policy.md [hunk 5]
```diff
@@ -3,123 +3,289 @@ title: AdGuard filter policy
...
-**What will be blocked with this filter?**
+#### AdGuard Widgets filter
 
-- This filter will block social media widgets (on third-party websites) such as "Like" and "Share" buttons, recommendation widgets, and more.
+This is a filter that blocks various widgets that are not essential for the functioning of the website or for interaction with it:
+
+- Widgets for content recommendations — related articles, similar websites, various personalized recommendations
+- Chat widgets that are not integrated with the content and are not the main content of the page
+- Marketing widgets:
+    - Chats for communication with assistants or bots
```
Comment (piquark6046) [inline docs/general/ad-filtering/filter-policy.md:198]: How can we handle the following case?
There is a chat widget. And, the widget is used at unrelated webpage and related webpage of a website.

Should be handled with `$path` modifier?

File: docs/general/ad-filtering/filter-policy.md (hunk 4 above)
Comment (Alex-302) [reply docs/general/ad-filtering/filter-policy.md:49]: > If there is a website that uses a language 1 and an advert issue exists with only its mobile version, can the issue be addressed in the language-specific filters list?

Mobile filter

> I think that AdGuard Mobile Ads filters list is too huge for some users group that visits only some regional websites.
Therefore, I think that we need to optimize the filters list.

It has only 6.7k rules.

> I face some such complaints from South Korean users.

about what?

File: docs/general/ad-filtering/filter-policy.md (hunk 5 above)
Comment (Alex-302) [reply docs/general/ad-filtering/filter-policy.md:198]: If annoying widget - we just block it on whole site.

**Review by vozersky — APPROVED** [review]

**Review by Alex-302 — APPROVED** [review]

File: docs/general/ad-filtering/filter-policy.md (hunk 4 above)
Comment (piquark6046) [reply docs/general/ad-filtering/filter-policy.md:49]: > about what?

Comment (piquark6046) [reply docs/general/ad-filtering/filter-policy.md:49]: > about what?

The fine delay during connecting a website.

**Comment by Yuki2718** [comment]
It may be too late to join the party, but the latter part of
> JavaScript rules should be used as sparingly as possible and only if blocking ads is impossible without them

sounds too strict. For example, we may add JS rule even when it's not necessary if the site in question cirumvents blocker regularly. I suggest to change `impossible` to `difficult`.

File: docs/general/ad-filtering/filter-policy.md (hunk 4 above)
Comment (Yuki2718) [reply docs/general/ad-filtering/filter-policy.md:49]: I doubt his/her diagnosis, even if it's true that will unlikely to be due to those regional rules. Such delay is usually caused by only one or a few probelmatic rules. The number of rules is not much releavant except for generic regex and cosmetic filters.

**Comment by Alex-302** [comment]
@Yuki2718 This phrase does not prohibit the use of JS rules (non-scriptlet).

**Comment by Yuki2718** [comment]
> This phrase does not prohibit the use of JS rules (non-scriptlet).

Then terminology must be more clear.

**Comment by Alex-302** [comment]
In our KB they are different rules
https://adguard.com/kb/general/ad-filtering/create-own-filters/#javascript-rules
https://adguard.com/kb/general/ad-filtering/create-own-filters/#scriptlets

**Pull request merged**